config/config.go

```diff
@@ -77,6 +78,13 @@ type Config struct {
 	toCache []byte
 }
 
+// MaintenanceConfig specifies a reconfigure sensor
+// When the sensor is read that value will be used to determine if it is safe to reconfigure.
+type MaintenanceConfig struct {
+	SensorName            string `json:"sensor_name"`
```

Comment:
Should this be a `resource.Name` instead of a `string`? I think will get rid of some proto conversion logic and error handling.

Reply:
Me and @cheukt just had a call about this and settled on keeping it a string. 

Changing it to a resource name leads to a bunch of additional complexity around validating the config when marshaling and unmarshaling.

Reply:
ya, an issue we've found with making it a `resource.Name` is that it ends up causing a lot of issues in the unmarshalling step if the name is malformed, which is difficult to control for. We'll have to figure out a good way to surface those errors as well. Since we have to do a bunch of validation/checking in `local_robot` anyway, it makes sense to just defer all the validation to that point.

definitely open to changing the type later, but didn't want to spend too long making this part work perfectly

Reply:
SGTM.